semanticdb/semanticdb3/guide.md

```diff
@@ -0,0 +1,576 @@
+    <td>
+      Specifies whether to include source code in
+      <code>TextDocument.text</code> (<code>fat</code> for yes,
+      <code>slim</code> for no).
+    </td>
+    <td><code>fat</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:failures</code></td>
+    <td>Enumeration</td>
+    <td>
+      Tells the Scala compiler how to report failures that may arise during
+      SemanticDB generation (<code>error</code>, <code>warning</code>,
+      <code>info</code>, <code>ignore</code>).
+    </td>
+    <td><code>warning</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:symbols</code></td>
+    <td>Enumeration</td>
+    <td>
+      Specifies what symbols will appear <code>TextDocument.symbols</code>
+      (<code>none</code> for none,
+      <code>definitions</code> for just symbols defined in the current program,
+      <code>all</code> for symbols either defined or referenced from the current
+      program).
+    </td>
+    <td><code>definitions</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:types</code></td>
+    <td>Enumeration</td>
+    <td>
+      Says whether to save type signatures in
+      <code>SymbolInformation.tpe</code> (<code>all</code> for yes,
+      <code>none</code> for no).
+    </td>
+    <td><code>all</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:profiling</code></td>
+    <td>Enumeration</td>
+    <td>
+      Controls basic profiling functionality that computes the overhead of
+      SemanticDB generation relative to regular compilation time
+      (<code>console</code> for dumping profiling information to console,
+      <code>off</code> for disabling profiling).
+    </td>
+    <td><code>off</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:include</code></td>
+    <td>Java regex</td>
+    <td>Which source files to include in SemanticDB generation?</td>
+    <td><code>.*</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:exclude</code></td>
+    <td>Java regex</td>
+    <td>Which source files to exclude from SemanticDB generation?</td>
+    <td><code>^$</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:diagnostics</code></td>
+    <td>Enumeration</td>
+    <td>
+      Says whether to save compiler messages in
+      <code>TextDocument.diagnostics</code> (<code>all</code> for yes,
+      <code>none</code> for no).
+    </td>
+    <td><code>all</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:synthetics</code></td>
+    <td>Enumeration</td>
+    <td>
+      Specifies whether to save compiler-generated code in
+      <code>TextDocument.synthetics</code> (<code>all</code> for yes,
+      <code>none</code> for no).
+    </td>
+    <td><code>all</code></td>
+  </tr>
+  <tr>
+    <td><code>-P:semanticdb:owners</code></td>
+    <td>Enumeration</td>
+    <td>
+      Specifies whether to save <code>SymbolInformation.owner</code>
+      (<code>all</code> for yes, <code>none</code> for no).
+    </td>
+    <td><code>all</code></td>
+  </tr>
+</table>
```

Comment:
Should we include examples? For example:

```
-P:semanticdb:synthetics:none # Don't emit synthetics
-P:semanticdb:mode:slim       # Don't include full text contents of source files
```

It's not clear from the table above that the values come after a colon with the syntax `-P:semanticdb:<flag>:<value>`